examples/vhdl/external_buffer/src/main.c

```diff
@@ -0,0 +1,66 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+extern int ghdl_main (int argc, char **argv);
+
+uint8_t *D[1];
+const uint32_t length = 5;
+
+void set_string_ptr(uint8_t id, uint8_t *p) {
+  //printf("C set_string_ptr(%d, %p)\n", id, p);
+  D[id] = p;
+}
+
+uintptr_t get_string_ptr(uint8_t id) {
+  //printf("C get_string_ptr(%d): %p\n", id, D[id]);
+  return (uintptr_t)D[id];
+}
+
+void write_char(uint8_t id, uint32_t i, uint8_t v ) {
+  //printf("C write_char(%d, %d): %d\n", id, i, v);
+  D[id][i] = v;
+}
+
+uint8_t read_char(uint8_t id, uint32_t i) {
+  //printf("C read_char(%d, %d): %d\n", id, i, D[id][i]);
+  return D[id][i];
+}
+
+static void exit_handler(void) {
+  int i, j, z, k;
+  uint8_t e, d;
+  k = 0;
+  for (j=0; j<3; j++) {
+    k += j;
+    for(i=0; i<length; i++) {
+      e = (i+1)*11 + k;
+      z = (length*j)+i;
+      d = D[0][z];
+      if (e != d) {
+        printf("check error %d: %d %d\n", z, e, d);
```

Comment:
It is helpful to distinguish between "got" and "expected" when printing these kinds of errors.

Reply:
Done.